pkg/http/client.go

```diff
@@ -76,22 +76,22 @@ func NewHttpClient(config cfg.Config, logger mon.Logger) Client {
 	settings.RetryCount = config.GetInt("http_client_retry_count")
 	settings.Timeout = config.GetDuration("http_client_request_timeout")
 
-	return NewHttpClientWithInterfaces(logger, mo, settings)
-}
-
-func NewHttpClientWithInterfaces(logger mon.Logger, mo mon.MetricWriter, settings *Settings) Client {
 	httpClient := resty.New()
 	httpClient.SetRedirectPolicy(resty.FlexibleRedirectPolicy(10))
 	httpClient.SetRetryCount(settings.RetryCount)
 	httpClient.SetTimeout(settings.Timeout)
 	httpClient.SetRetryWaitTime(settings.RetryWaitTime)
 	httpClient.SetRetryMaxWaitTime(settings.RetryMaxWaitTime)
 
+	return NewHttpClientWithInterfaces(logger, mo, httpClient)
+}
+
+func NewHttpClientWithInterfaces(logger mon.Logger, mo mon.MetricWriter, httpClient *resty.Client) Client {
```

Comment:
What's your use case here? `httpClient *resty.Client`: I don't see the benefit of directly passing the client to the test method..

Reply:
sorry, as said should have been an interface. Was just too sleepy to realize i was passing a pointer